app/commands/thredded/autofollow_users.rb

```diff
@@ -1,18 +1,23 @@
 # frozen_string_literal: true
 module Thredded
-  class AutofollowMentionedUsers
+  class AutofollowUsers
     def initialize(post)
       @post = post
     end
 
     def run
       autofollowers.each do |user|
-        Thredded::UserTopicFollow.create_unless_exists(user.id, post.postable_id, :mentioned)
+        reason = mentioned_users.include?(user) ? :mentioned : :auto
+        Thredded::UserTopicFollow.create_unless_exists(user.id, post.postable_id, reason)
       end
     end
 
+    def mentioned_users
+      @mentioned_users ||= Thredded::AtNotificationExtractor.new(post).run
+    end
+
     def autofollowers
-      autofollowers = Thredded::AtNotificationExtractor.new(post).run
+      autofollowers = (include_auto_followers + mentioned_users).uniq
       autofollowers.delete(post.user)
       exclude_those_opting_out_of_at_notifications autofollowers
```

Comment:
The users that don't have read permissions for the topic should also be excluded (otherwise we may leak the contents in the notification email).